clang/test/CodeGen/debug-dir-win-pe-ltcg-string.c

```diff
@@ -0,0 +1,13 @@
+// This test checks if Window PE file compiled with -flto option contains a magic 
+// string "LTCG" to indicate LTO compilation.
+
+// REQUIRES: system-windows
+
+// RUN: %clang --target=x86_64-pc-windows-msvc -flto -fuse-ld=lld %s -o %t.exe
+// RUN: dumpbin /HEADERS %t.exe | FileCheck %s
```

Comment:
I don't think I am sure it's correct to rely on dumpbin here. I think we have other tools in LLVM that can extract this info and it would allow us to run this test on another OS as well.

Reply:
I was using `llvm-objdump` while I was working on that, but I think that using `dumpbin` is better compared to that. It can be somewhat tricky to emit this info in such a way that `dumpbin` catches it: first this info has to be emitted in the debug dir and in such a way that `dumpbin` recognizes the string (I believe it's length has to be divisible by 4 -- if not it has to prefixed with null bytes). Using the `dumpbin` to test assures that it correctly sees the magic strings. Checking for simple existence of string in the Windows PE file, using `llvm-objdump` for example, is not enough, since the string can have wrong length, and not be recognized by `dumpbin`. I was also aiming to achieve running on other OS, but that means dropping the `dumpbin`, which I think is wrong and the "PGU" test still has to be run on Windows (since the binary has to be run)

Reply:
@mikolaj-pirog One of the goals of LLVM
is cross compilation. `dumpbin` will not be available on Linux. We usually use `REQUIRES: system-windows` for testing low-level stuff on that system. For this case you’d probably want to change `llvm-objdump` to match `dumpbin` behavior.

Reply:
> @mikolaj-pirog One of the goals of LLVM is cross compilation. `dumpbin` will not be available on Linux. We usually use `REQUIRES: system-windows` for testing low-level stuff on that system. For this case you’d probably want to change `llvm-objdump` to match `dumpbin` behavior.
Okay, I think I will do the following: I will change the tests to use `llvm-objdump` with `-s` option to simply check for presence of the  magic strings, for now accepting the problems mentioned above. After this patch is merged, I will prepare new patch introducing an option to `llvm-objdump` to display contents of debug dir, the way `dumpbin` does, and correct the tests. Would that be OK?

Reply:
Yes that sounds good.